packages/aws-cdk-lib/aws-route53/lib/health-check.ts

```diff
@@ -0,0 +1,517 @@
+   *
+   * @default 30
+   */
+  readonly requestInterval?: Duration;
+
+  /**
+   * The path that you want Amazon Route 53 to request when performing health checks. The path can be any value for which your endpoint will return an HTTP status code of 2xx or 3xx when the endpoint is healthy, for example the file /docs/route53-health-check.html. Route 53 automatically adds the DNS name for the service and a leading forward slash (/) character.
+   *
+   * @default - not configured
+   */
+  readonly resourcePath?: string;
+
+  /**
+   * The Amazon Resource Name (ARN) of the Route 53 Application Recovery Controller routing control that you want Amazon Route 53 health checkers to use to determine whether the specified health check is healthy.
+   *
+   * @default - not configured
+   */
+  readonly routingControl?: string;
+
+  /**
+   * The string that you want Amazon Route 53 to search for in the response body from the specified resource. If the string appears in the response body, Route 53 considers the resource healthy.
+   *
+   * Route 53 considers case when searching for SearchString in the response body.
+   *
+   * @default - not configured
+   */
+  readonly searchString?: string;
+}
+
+/**
+ * Amazon Route 53 health checks monitor the health and performance of your web applications, web servers, and other resources. Each health check that you create can monitor one of the following:
+ * - The health of a resource, such as a web server,
+ * - The status of other health checks, and
+ * - The CloudWatch alarm that you specify,
+ * - The status of an Amazon Route 53 routing control.
+ */
+export class HealthCheck extends Resource implements IHealthCheck {
+  /**
+   * Import an existing health check into this CDK app.
+   *
+   * @param scope The parent creating construct (usually `this`).
+   * @param id The construct's name.
+   * @param healthCheckId ID of the health check.
+   * @returns a reference to the existing health check.
+   */
+  public static fromHealthCheckId(scope: Construct, id: string, healthCheckId: string): IHealthCheck {
+    class Import extends Resource implements IHealthCheck {
+      public readonly healthCheckId = healthCheckId;
+    }
+
+    return new Import(scope, id);
+  }
+
+  public readonly healthCheckId: string;
+
+  /**
+   * Creates a new health check.
+   *
+   * @param scope The parent creating construct (usually `this`).
+   * @param id The construct's name.
+   * @param props the properties of the new health check.
+   * @returns a reference to the newly created health check.
+   */
+  constructor(scope: Construct, id: string, props: HealthCheckProps) {
+    super(scope, id);
+
+    validateProperties(props);
+
+    const resource = new CfnHealthCheck(this, 'Resource', {
+      healthCheckConfig: {
+        type: props.type,
+        alarmIdentifier: props.alarmIdentifier,
+        childHealthChecks: props.childHealthChecks,
```

Comment:
Even though this is not well documented, I'm assuming that `childHealthChecks` is a list of `Ref`s/`healthCheckId`s. I would add an integration test to verify this assumption, and change this type from `string[]` to `IHealthCheck[]`. CDK users will then be able to create multiple health checks in the same stack, or import them with `fromHealthCheckId`.

This should also be able to handle dependency resolution (e.g. if you create both `hc1` and `hc2`, with `hc2` being a child of `hc1`,) but again, that's something you can verify with an integration test